autotest/ogr/ogr_mssqlspatial.py

```diff
@@ -711,6 +736,35 @@ def test_binary_field_bcp():
 #
 
 
+def test_geometry_column_identification():
+    """Test for issue GH #1190
+    SQL Server data source: geometry column not identified by ogr2ogr"""
+
+    if gdaltest.mssqlspatial_ds is None:
+        pytest.skip("MSSQLSpatial driver not available")
+
+    gdaltest.mssqlspatial_ds.ExecuteSQL(
+        "SELECT *, ogr_geometry.STCentroid() AS shape INTO dbo.poly_shape FROM dbo.tpoly"
+    )
+
+    try:
+        ds = ogr.Open(gdaltest.mssqlspatial_dsname)
+        lyr = ds.ExecuteSQL("SELECT eas_id, shape FROM poly_shape")
```

Comment:
cf above mention about ExecuteSQL() leaks

Reply:
I am missing something: when ds and lyr go out of scope doesn't python swig call the destructors and free memory?

Besides that when the python test ends memory will be reclaimed anyway.

Reply:
> when ds and lyr go out of scope doesn't python swig call the destructors and free memory?

Not for a ExecuteSQL() layer that must be explicitly freed: https://gdal.org/api/python/osgeo.gdal.html#osgeo.gdal.Dataset.ExecuteSQL  . Perhaps this could be changed but this would have implications probably on the C++ side of things, probably in all drivers that implement ExecuteSQL()

> Besides that when the python test ends memory will be reclaimed anyway.

no, the C++ layer woud be leaking.